scarlet/interpolation.py

```diff
@@ -419,3 +419,52 @@ def sinc2D(y, x):
         2-D sinc evaluated in x and y
     '''
     return np.dot(np.sinc(y), np.sinc(x))
+
+
+def subsample_function(y, x, f, dNy, dNx=None, dy=None, dx=None):
+    """Subsample a function
+    Given the expected pixel grid of a function, subsample that function
+    at a grid subdivided in x by `dNx` and y by `dNy`.
+    """
+    # Use the spacing between x values to define the subsampled regions
+    if dx is None:
+        dx = x[1] - x[0]
+    if dy is None:
+        dy = y[1] - y[0]
+    if dNx is None:
+        dNx = dNy
+    assert dNy % 2 == 0, "dNy must be even, received {0}".format(dNy)
+    assert dNx % 2 == 0, "dNx must be even, received {0}".format(dNx)
+    assert np.all(np.isclose(x[1:]-x[:-1], x[1] - x[0])), "x must have equal spacing"
+    assert np.all(np.isclose(y[1:]-y[:-1], y[1] - y[0])), "y must have equal spacing"
+
+    # Create the subsampled interval and use it to sample `f`
+    _x = np.linspace(x[0]-dx/2, x[-1]+dx/2, len(x)*dNx+1)
+    _y = np.linspace(y[0]-dy/2, y[-1]+dy/2, len(y)*dNy+1)
+    return f(_y, _x), _y, _x
+
+
+def apply_2D_trapezoid_rule(y, x, f, dNy, dNx=None, dy=None, dx=None):
+    """Use the trapezoid rule to integrate over a subsampled function
+    2D implementation of the trapezoid rule.
+    See `apply_trapezoid_rule` for a description, with the difference
+    that `f` is a function `f(y,x)`, where we note the c ++`(y,x)` ordering.
+    """
```

Comment:
A miny tiny little comment. I would prefer if we put operations over `y` berfore the corresponding operation over `x`, just to remain consistent with the convention. I found the ordering helped avoiding confusion.